test/notifications/display_test.dart

```diff
@@ -208,6 +211,158 @@ void main() {
         ..vibrationPattern.isNotNull().deepEquals(
             NotificationChannelManager.kVibrationPattern);
     });
+
+    test('on Android 28 (and lower) resource file is used for notification sound', () async {
+      addTearDown(testBinding.reset);
+      final androidNotificationHost = testBinding.androidNotificationHost;
+
+      // Override android version
+      testBinding.deviceInfoResult =
+        const AndroidDeviceInfo(sdkInt: 28, release: '10');
+
+      // Ensure that on Android 10, notification sounds aren't being copied to
+      // the media store, and resource file is used directly.
+      await NotificationChannelManager.ensureChannel();
+      check(androidNotificationHost.takeCopySoundResourceToMediaStoreCalls()).length.equals(0);
+
+      final defaultSoundResourceName =
+        NotificationChannelManager.kDefaultNotificationSound.resourceName;
+      final soundUrl =
+        'android.resource://com.zulip.flutter/raw/$defaultSoundResourceName';
+      check(androidNotificationHost.takeCreatedChannels()).single
+        ..id.equals(NotificationChannelManager.kChannelId)
+        ..name.equals('Messages')
+        ..importance.equals(NotificationImportance.high)
+        ..lightsEnabled.equals(true)
+        ..soundUrl.equals(soundUrl)
+        ..vibrationPattern.isNotNull().deepEquals(
+            NotificationChannelManager.kVibrationPattern);
```

Comment:
```suggestion
      check(androidNotificationHost.takeCreatedChannels()).single
        ..soundUrl.equals(soundUrl);
```

This is the only part of this check that differs between these new test cases, or between them and the corresponding check in the smoke test, right?

The rest of the details of this call aren't what these test cases are meant to focus on. So leaving them out helps the reader focus their attention on the part that the test is about.